src/main/scala/com/microsoft/hyperspace/actions/RefreshActionBase.scala

```diff
@@ -64,13 +64,23 @@ private[actions] abstract class RefreshActionBase(
 
   // Reconstruct a df from schema
   protected lazy val df = {
-    val rels = previousIndexLogEntry.relations
-    val dataSchema = DataType.fromJson(rels.head.dataSchemaJson).asInstanceOf[StructType]
-    spark.read
-      .schema(dataSchema)
-      .format(rels.head.fileFormat)
-      .options(rels.head.options)
-      .load(rels.head.rootPaths: _*)
+    val rel = previousIndexLogEntry.relations.head
+    val dataSchema = DataType.fromJson(rel.dataSchemaJson).asInstanceOf[StructType]
+    if (rel.fileFormat.equals("delta")) {
+      // Exclude "versionAsOf" and "timestampAsOf" options, so that this DataFrame for refresh
+      // can see the latest snapshot.
```

Comment:
does it mean that user cannot create index on a previous version?

Reply:
No, but this is the refresh action so that we should remove the versionAsOf or timestampAsOf option to refresh based on the latest version. 
If a user would like to refresh with a specific version, not the latest version, then we might need to extend the refresh API.